pkg/apis/openebs.io/scp/v1alpha1/storage-pool.go

```diff
@@ -29,8 +31,345 @@ import (
+	Total resource.Quantity `json:"total,omitempty"`
+
+	// Used Capacity of the pool
+	Used resource.Quantity `json:"used,omitempty"`
+
+	// Available Capacity of the pool
+	Available resource.Quantity `json:"available,omitempty"`
+}
+
+// StorageIOPs for the pool,viz total, provisioned, used and available capacity
+type StorageIOPs struct {
+	// Total IOPs of the pool
+	Total uint64 `json:"total,omitempty"`
+
+	// Available IOPs of the pool
+	Available uint64 `json:"available,omitempty"`
+
+	// Provisioned IOPs of the pool
+	Provisioned uint64 `json:"provisioned,omitempty"`
+
+	// Used IOPs of the pool
+	Used uint64 `json:"used,omitempty"`
+}
+
+type DeviceConfiguration struct {
+	// Devices for the list of UIDs of the devices
+	Devices []types.UID `json:"devices,omitempty"`
+
+	// RaidConfiguration for the raid configurations currently being used
+	// +nullable
+	RaidConfiguration *map[string]int `json:"raidConfiguration,omitempty"`
+}
+
+// StoragePoolState for the state of the pool for various scenarios
+type StoragePoolState string
+
+const (
+	// StoragePoolStatusEmpty ensures the create operation is to be done, if import fails.
+	StoragePoolStatusEmpty StoragePoolState = ""
+
+	// StoragePoolStatusOnline signifies that the pool is online.
+	StoragePoolStatusOnline StoragePoolState = "Online"
+
+	// StoragePoolStatusOffline signifies that the pool is offline.
+	StoragePoolStatusOffline StoragePoolState = "Offline"
+
+	// StoragePoolStatusDegraded signifies that the pool is degraded.
+	StoragePoolStatusDegraded StoragePoolState = "Degraded"
+
+	// StoragePoolStatusFaulted signifies that the pool is faulted.
+	StoragePoolStatusFaulted StoragePoolState = "Faulted"
+
+	// StoragePoolStatusRemoved signifies that the pool is removed.
+	StoragePoolStatusRemoved StoragePoolState = "Removed"
+
+	// StoragePoolStatusUnavailable signifies that the pool is not available.
+	StoragePoolStatusUnavailable StoragePoolState = "Unavailable"
+
+	// StoragePoolStatusError signifies that the pool status could not be fetched.
+	StoragePoolStatusError StoragePoolState = "Error"
+
+	// StoragePoolStatusDeletionFailed ensures the resource deletion has failed.
+	StoragePoolStatusDeletionFailed StoragePoolState = "DeletionFailed"
+
+	// StoragePoolStatusInvalid ensures invalid resource.
+	StoragePoolStatusInvalid StoragePoolState = "Invalid"
+
+	// StoragePoolStatusErrorDuplicate ensures error due to duplicate resource.
+	StoragePoolStatusErrorDuplicate StoragePoolState = "ErrorDuplicate"
+
+	// StoragePoolStatusPending ensures pending task for StoragePool.
+	StoragePoolStatusPending StoragePoolState = "Pending"
+
+	// StoragePoolStatusInit is initial state of StoragePool, before pool creation.
+	StoragePoolStatusInit StoragePoolState = "Init"
+
+	// StoragePoolStatusCreateFailed is state when pool creation failed
+	StoragePoolStatusCreateFailed StoragePoolState = "PoolCreationFailed"
+)
+
+// StoragePoolMode for the storage pool
+type StoragePoolMode string
```

Comment:
This seems to be lvm-specific configuration for now.

Reply:
Removing it.